osu.Game/Online/OnlineViewContainer.cs

```diff
@@ -0,0 +1,115 @@
+        protected override Container<Drawable> Content => View;
+
+        [Resolved]
+        protected IAPIProvider API { get; private set; }
+
+        public OnlineViewContainer(string placeholderMessage)
+        {
+            InternalChildren = new Drawable[]
+            {
+                View = new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                },
+                placeholderContainer = new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Alpha = 0,
+                    Child = placeholder = new LoginPlaceholder($@"Please sign in to {placeholderMessage}")
+                },
+                loading = new LoadingAnimation
+                {
+                    Alpha = 0,
+                }
+            };
+        }
+
+        public virtual void APIStateChanged(IAPIProvider api, APIState state)
+        {
+            switch (state)
+            {
+                case APIState.Failing:
+                case APIState.Connecting:
+                    Schedule(() => UpdatePlaceholderVisibility(PlaceholderStatus.Connecting));
+                    break;
+
+                case APIState.Offline:
+                    Schedule(() => UpdatePlaceholderVisibility(PlaceholderStatus.Offline));
+                    break;
+
+                case APIState.Online:
+                    Schedule(() => UpdatePlaceholderVisibility(PlaceholderStatus.Online));
+                    break;
+            }
+        }
+
+        protected void UpdatePlaceholderVisibility(PlaceholderStatus status)
+        {
+            switch (status)
+            {
+                case PlaceholderStatus.Offline:
+                    View.FadeOut(150, Easing.OutQuint);
+                    placeholder.ScaleTo(0.8f).Then().ScaleTo(1, 3 * transform_time, Easing.OutQuint);
+                    placeholderContainer.FadeInFromZero(2 * transform_time, Easing.OutQuint);
+                    loading.Hide();
+                    break;
+
+                case PlaceholderStatus.Online:
+                    placeholderContainer.FadeOut(150, Easing.OutQuint);
+                    View.FadeIn(transform_time, Easing.OutQuint);
+                    loading.Hide();
+                    break;
+
+                case PlaceholderStatus.Connecting:
+                    loading.Show();
+                    placeholderContainer.FadeOut(150, Easing.OutQuint);
+                    View.FadeOut(150, Easing.OutQuint);
+                    break;
+            }
+        }
+
+        protected override void LoadComplete()
+        {
+            API?.Register(this);
+            base.LoadComplete();
+        }
+
+        protected override void Dispose(bool isDisposing)
+        {
+            API?.Unregister(this);
+            base.Dispose(isDisposing);
+        }
+
+        protected enum PlaceholderStatus
+        {
+            Offline,
+            Online,
+            Connecting,
```

Comment:
The namings in this enum seem out-of-scope, what you probably want here is:
 - `Visible`
 - `Hidden`
 - `Loading`

with xmldocs shortly explaining what these status mean to the container.

Reply:
Reason for this is to expand the container into supporting more than just 1 (not logged in) placeholder.